bitcoin/src/crypto/key.rs

```diff
@@ -1036,7 +1036,7 @@ impl fmt::Display for ParsePublicKeyError {
         use ParsePublicKeyError::*;
         match self {
             Encoding(e) => write_err!(f, "string error"; e),
-            InvalidChar(char) => write!(f, "hex error {}", char),
+            InvalidChar(err) => write!(f, "hex error {} at position {}", err.invalid_char() as char, err.pos()),
```

Comment:
In de02371cf66428d92d5db952ce27b97d23cfaa2c:

I think we should check `invalid_char` to decide if it's printable (unless I'm crazy there's no stdlib function for this, so just check whether it's between `b' '` and `b'~'`) and use char::from if so. Otherwise output it in hex.

Reply:
If we want to do that we should do it in `hex` crate not here, right? This crate should just pass through to the `Display` impl of `hex::InvalidCharError` using `write_err!` like we normally do.

Reply:
I would have expected to see 
```rust
        match *self {
            Encoding(ref e) => write_err!(f, "string error"; e),
            InvalidChar(ref e) => write_err!(f, "hex decoding"; e),
            InvalidHexLength(got) =>
                write!(f, "pubkey string should be 66 or 130 digits long, got: {}", got),
        }
```

Reply:
Yep, agreed.